[ao][sparsity] add option for tensor_fqn to sparsity API

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79774
* #79598
* #79597
* __->__ #79113

Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

HDCharles committed:
[WIP][ao][sparsity] add option for tensor_fqn to sparsity API

Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[WIP][ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

z-a-f commented on test/ao/sparsity/test_pruner.py:
```diff
@@ -238,7 +238,7 @@ def _check_pruner_valid_before_step(self, model, pruner, device):
 
     def _check_pruner_valid_after_step(self, model, pruner, pruned_set, device):
-        for config in pruner.module_groups:
+        for config in pruner.tensor_groups:
```
wdyt, if we just use "groups"? Or is it too confusing?

HDCharles replied:
sure, I can do that.

z-a-f reviewed (approved):
lgtm

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] add option for tensor_fqn to sparsity API"


Summary: updated sparsity api to accept tensor_fqn as primary
specification method, i.e. [{'tensor_fqn':
'linear.weight'}]

Pruning API also updated due to knock on changes.

left old api for accepting module_fqns but changed 'fqn' to 'module_fqn'
for clarity (this will break BC)

updated variables in code to use module rather than layer

updated state dict to use tensor_fqn rather than 'fqn' or 'module_fqn'

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
